client/src/containers/KsqlDB/KsqlDBList/KsqlDBList.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Header from '../../Header/Header';
+import { getSelectedTab } from '../../../utils/functions';
+import { Link } from 'react-router-dom';
+import KsqlDBInfo from './KsqlDBInfo/KsqlDBInfo';
+import KsqlDBStreams from './KsqlDBStreams/KsqlDBStreams';
+import KsqlDBTables from './KsqlDBTables/KsqlDBTables';
+import KsqlDBQueries from './KsqlDBQueries/KsqlDBQueries';
+
+class KsqlDBList extends Component {
+  state = {
+    clusterId: this.props.history.clusterId || this.props.match.params.clusterId,
+    ksqlDBId: this.props.history.ksqlDBId || this.props.match.params.ksqlDBId,
+    selectedTab: 'streams',
+    roles: JSON.parse(sessionStorage.getItem('roles')),
+  };
+
+  tabs = ['streams', 'tables', 'queries', 'info'];
+
+  componentDidMount() {
+    const { clusterId, ksqlDBId } = this.props.match.params;
+    const tabSelected = getSelectedTab(this.props, this.tabs);
+    this.setState({ selectedTab: tabSelected ? tabSelected : 'streams' }, () => {
+      this.props.history.replace(
+        `/ui/${clusterId}/ksqldb/${ksqlDBId}/${this.state.selectedTab}`
+      );
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.location.pathname !== prevProps.location.pathname) {
+      const tabSelected = getSelectedTab(this.props, this.tabs);
+      this.setState({ selectedTab: tabSelected });
+    }
+  }
+
+  tabClassName = tab => {
+    const { selectedTab } = this.state;
+    return selectedTab === tab ? 'nav-link active' : 'nav-link';
+  };
+
+  renderSelectedTab() {
+    const { clusterId, ksqlDBId, selectedTab } = this.state;
+    const { history, match, location } = this.props;
+
+    switch (selectedTab) {
```

Comment:
If you introduce a little mapping like:

```jsx
const tabs = {
  info: KsqlDBInfo,
  streams: KsqlDBStreams,
  tables: KsqlDBTables,
  queries: KsqlDBQueries
};
```

Then you can get rid of the switch all together like this:

```jsx
    const SelectedTab = tabs[selectedTab] || KsqlDBStreams;

    return (
      <SelectedTab
        clusterId={clusterId}
        ksqlDBId={ksqlDBId}
        history={history}
        match={match}
        location={location}
      />
    );
```

Probably a bit of a personal preference but I think the version with the map leads to a little more maintainable code (and it's a little shorter too)

Reply:
I would have done the same! I'll go over the kafka connect code soon-ish and see what I can do